### PR TITLE
Multi-step model with walk-forward

### PR DESCRIPTION
**Pull request by sarajcev**
Multi-step predictions by using a walk-forward approach.
[Walk-forward predictions](https://app.gitkraken.com/glo/card/c52efb12ec904cd29d5f303e7e0cf1e1)

### CODE REVIEW

File: model.ipynb [hunk 1]
```diff
@@ -35,7 +35,6 @@
    "outputs": [],
```
Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025190907760953641992 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 23px; height: 21px; left: 114px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025191501690995311755 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 7px; height: 21px; left: 61px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025191507712737708407 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 26px; height: 21px; left: 72px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025191524078038748719 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 39px; height: 21px; left: 102px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025192678744943659360 gwmwi-0 ginger-module-highlighter-mistake-anim no-suggestion" style="position: absolute; top: 12px; width: 56px; height: 21px; left: 80px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025194029686678164439 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 25px; height: 21px; left: 72px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025194060978683821612 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 44px; height: 21px; left: 101px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025194099623542397491 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 45.7344px; height: 21px; left: 149px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 314px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025197716988932822575 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 24px; height: 21px; left: 561px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025197716987038328382 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 33px; height: 21px; left: 596px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025197716986350207263 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 31px; width: 14px; height: 21px; left: 509px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (antunmeg) [inline model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025198911110321736979 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 15px; height: 21px; left: 81px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025198927412858595516 gwmwi-0 ginger-module-highlighter-mistake-anim no-suggestion" style="position: absolute; top: 12px; width: 26px; height: 21px; left: 15px;"></gwmw></gdiv>
<br/>

 _Reply via <a href='https://app.reviewnb.com/sarajcev/Seminar/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (sarajcev) [reply model.ipynb:35]: Ne vidim komentare? Niti u prethodnim slučajevima koje sam potvrdio. Da li ste nešto htjeli komentirati?

Comment (antunmeg) [reply model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025285635907150987461 gwmwi-0 ginger-module-highlighter-mistake-anim no-suggestion" style="position: absolute; top: 12px; width: 11px; height: 21px; left: 54px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-2 gwmw-16025285635902613122701 gwmwi-0 ginger-module-highlighter-mistake-anim" style="position: absolute; top: 12px; width: 8px; height: 21px; left: 180px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025285635903882123933 gwmwi-0 ginger-module-highlighter-mistake-anim no-suggestion" style="position: absolute; top: 12px; width: 15px; height: 21px; left: 312px;"></gwmw><gwmw class="ginger-module-highlighter-mistake-type-1 gwmw-16025285635904278132077 gwmwi-0 ginger-module-highlighter-mistake-anim no-suggestion" style="position: absolute; top: 12px; width: 41px; height: 21px; left: 331px;"></gwmw></gdiv>

Comment (antunmeg) [reply model.ipynb:35]: <p>Komentar</p>

Comment (antunmeg) [reply model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"></gdiv>

Comment (antunmeg) [reply model.ipynb:35]: <p>Ovdje bi sve varijable i komentare preveo na engleski jezik.</p>

Comment (antunmeg) [reply model.ipynb:35]: <p>Pokušao sam koristiti joblib na staroj verziji koda. Međutim, umjesto 11-13 minuta koliko je trajala simulacija bez jobliba na mom računalu, vrijeme se povećalo na 14 min i 20 sekundi!</p><p>Joblib sam uključio pomoću memory parametra prema dokumentaciji <a href="https://scikit-learn.org/stable/auto_examples/compose/plot_compare_reduction.html" target="_blank">https://scikit-learn.org/stable/auto_examples/compose/plot_compare_reduction.html</a></p>

Comment (antunmeg) [reply model.ipynb:35]: <p>Evo i output:</p><pre class="ql-syntax" spellcheck="false">[Memory] Calling sklearn.pipeline._fit_transform_one...
_fit_transform_one(StandardScaler(copy=True, with_mean=True, with_std=True), PV_1h PV_2h PV_3h PV_4h PV_5h \
2019-05-14 00:00:00+00:00 0.000000 0.000000 0.000000 0.000000 0.001337
2019-05-14 01:00:00+00:00 0.000000 0.000000 0.000000 0.000000 0.000000
2019-05-14 02:00:00+00:00 0.000000 0.000000 0.000000 0.000000 0.000000
2019-05-14 03:00:00+00:00 0.000000 0.000000 0.000000 0.000000 0.000000
2019-05-14 04:00:00+00:00 0.000585 0.000000 0.000000 0.000000 0.000000
... ... ... ... ... ...
2019-06-10 18:00:00+00:00 0.034669 0.022396 0.053528 0.175738 0.341665
2019-06-10 19:00:00+00:00 0.020568 0.034669 0.022396 0.053528 0...,
2019-05-14 00:00:00+00:00 0.000000
2019-05-14 01:00:00+00:00 0.000000
2019-05-14 02:00:00+00:00 0.000000
2019-05-14 03:00:00+00:00 0.000585
2019-05-14 04:00:00+00:00 0.017226
...
2019-06-10 18:00:00+00:00 0.020568
2019-06-10 19:00:00+00:00 0.004382
2019-06-10 20:00:00+00:00 0.000000
2019-06-10 21:00:00+00:00 0.000000
2019-06-10 22:00:00+00:00 0.000000
Freq: H, Name: PV, Length: 671, dtype: float64,
None, message_clsname='Pipeline', message=None)
________________________________________________fit_transform_one - 0.0s, 0.0min
________________________________________________________________________________
[Memory] Calling sklearn.pipeline._fit_transform_one...
_fit_transform_one(SelectKBest(k=52,
score_func=&lt;function mutual_info_regression at 0x00000225121D0948&gt;),
array([[-0.865122, ..., -1.513015],
...,
[-0.865122, ..., -1.513015]]),
2019-05-14 00:00:00+00:00 0.000000
2019-05-14 01:00:00+00:00 0.000000
2019-05-14 02:00:00+00:00 0.000000
2019-05-14 03:00:00+00:00 0.000585
2019-05-14 04:00:00+00:00 0.017226
...
2019-06-10 18:00:00+00:00 0.020568
2019-06-10 19:00:00+00:00 0.004382
2019-06-10 20:00:00+00:00 0.000000
2019-06-10 21:00:00+00:00 0.000000
2019-06-10 22:00:00+00:00 0.000000
Freq: H, Name: PV, Length: 671, dtype: float64,
None, message_clsname='Pipeline', message=None)
________________________________________________fit_transform_one - 3.4s, 0.1min
Execution time (hour:min:sec): 0:14:23.275570
Best parameter (CV score = -0.002):
{'forest__max_depth': None, 'forest__n_estimators': 516, 'kbest__k': 52, 'preprocess': StandardScaler(copy=True, with_mean=True, with_std=True)}
</pre>

Comment (antunmeg) [reply model.ipynb:35]: <gdiv class="ginger-module-highlighter-float-content" style="height: 278px;"></gdiv>

Comment (antunmeg) [reply model.ipynb:35]: <p>Ovdje je sve jasno.</p>

Comment (sarajcev) [reply model.ipynb:35]: Očito joblib stvara overhead pa ne doprinosi.

Comment (sarajcev) [reply model.ipynb:35]: Ne vidim na šta se odnosi ovaj komentar (nije povezan s kodom).

Comment (sarajcev) [reply model.ipynb:35]: Single-step s koračanjem. Terminologija je nejasna.

Comment (sarajcev) [reply model.ipynb:35]: Ovo je još jedan način da se cikličnost vremenske serije predstavi modelu. Hammingov prozor je najsličniji kosinusnom (koji nije implementiran u pandasu).

Comment (sarajcev) [reply model.ipynb:42]: Komentari su sada na engl.

**Comment by sarajcev** [comment]
@antunmeg Notebook je revidiran i djelomično unaprijeđen kroz nekoliko commits-a. PR je spreman za review.

**Pull request merged**